lib/ansible/modules/database/mysql/mysql_variables.py

```diff
@@ -177,7 +222,20 @@ def main():
         else:
             module.fail_json(msg="unable to find %s. Exception message: %s" % (config_file, to_native(e)))
 
-    mysqlvar_val = getvariable(cursor, mysqlvar)
+    mysqlvar_val = None
+    var_in_mysqld_auto_cnf = None
+
+    if mode in ('persist', 'persist_only'):
+        var_in_mysqld_auto_cnf = check_mysqld_auto(module, cursor, mysqlvar)
+
+    if mode in ('global', 'persist'):
+        mysqlvar_val = getvariable(cursor, mysqlvar)
+    elif mode == 'persist_only':
+        if var_in_mysqld_auto_cnf is None:
+            mysqlvar_val = False
+        else:
+            mysqlvar_val = var_in_mysqld_auto_cnf
+
     if mysqlvar_val is None:
```

Comment:
How about moving this in the main `if:` branch above? Then there's no need to do funny tricks for `mysqlvar_val` in the `elif:` branch.

Reply:
i changed the code but not in this part. Could you take another look and suggest a definite solution if it is still relevant

Reply:
@felixfontein , i think it should be rewritten totally with keeping existing interfaces and their behavior. I'm going to do this soon (+ check_mode support).
I believe all of strange things will be removed there.